tests/main/views/test_g_cloud.py

```diff
@@ -18,15 +17,18 @@ def teardown_method(self, method):
         self._search_api_client_patch.stop()
         super().teardown_method(method)
 
-    def test_renders_correct_search_links(self):
-        self._search_api_client.search.return_value = self.search_results
-
-        res = self.client.get("/buyers/direct-award/g-cloud/choose-lot")
-        assert res.status_code == 200
+    @pytest.mark.parametrize('lot', ('cloud-support', 'cloud-software', 'cloud-hosting', ''))
+    def test_posting_lot_redirects_to_search_with_lot(self, lot):
+        res = self.client.post(
+            "/buyers/direct-award/g-cloud/choose-lot",
+            data={"lot": lot}
+        )
 
-        body = res.get_data(as_text=True)
-        doc = html.fromstring(body)
+        assert res.status_code == 302
+        assert res.location.endswith(f"/g-cloud/search?lot={lot}")
 
-        assert doc.xpath("//form[@action=$u]", u="/g-cloud/search")
-        # at least one link into a specific lot
-        assert doc.xpath("//a[starts-with(@href, $u)]", u="/g-cloud/search?lot=")
+        self._search_api_client.search.return_value = self.search_results
+        self._search_api_client.aggregate.return_value = \
+            self._get_fixture_data('g9_aggregations_fixture.json')
```

Comment:
I put them half-way through because I wasn't sure what would happen if they were at the top, but I'll try it and if there are no ill effects I'll commit the change.